sailor/sap_iot/write.py

```diff
@@ -68,22 +70,82 @@ def _upload_data_single_indicator_group(dataset, indicator_set, group_id, templa
         _upload_data_single_equipment(data_subset, equipment.id, tags)
 
 
-def upload_indicator_data(dataset: TimeseriesDataset):
+def _check_indicator_group_is_complete(uploaded_indicators, indicator_group_id, template_id):
+    missing_indicators = []
+    request_url = _ac_application_url() + VIEW_TEMPLATES + '/' + template_id
+    oauth_ac = get_oauth_client('asset_central')
+    template = oauth_ac.request('GET', request_url)
+
+    if len(template) == 0:
+        raise RuntimeError(f' No template found for template id {template_id}.')
+
+    if len(template) > 1:
+        LOG.warning(f'More than one template found for template id {template_id}. Selecting first.')
+
+    item = template[0]
+    filtered_indicator_groups = list(filter(lambda x: x['id'] == indicator_group_id, item['indicatorGroups']))
+
+    if len(filtered_indicator_groups) == 0:
+        raise RuntimeError(f'Could not find an indicator group {indicator_group_id} for template {template_id}.')
+
+    if len(filtered_indicator_groups) > 1:
+        group_id = filtered_indicator_groups[0]['internalId'] + ', template: ' + template_id
```

Comment:
since this is specifically the case where you have more than one element in the `filtered_indicator_groups` extracting the `internalId` from that element could be misleading (another element might have another `internalId`). Better use the `indicator_group_id` in this message (even though that's ID not name).

Also a bit confused why you do the string interpolation here, and then again in line 93. Why not do everything in l 93? Calling the result of this string interpolation `group_id` if it contains the template plus some extra text is also weird.

Reply:
try to make it obvious that a key pair indicator group / template has several entries. which normally should not happen. try to make it more clear